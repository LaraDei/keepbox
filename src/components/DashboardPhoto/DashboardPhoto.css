.backbtn{
    background-color: #cacaca;
    border-radius: 30px;
    border: 0;
    color: black;
    font-size: 12px;
    font-weight: bold;
    height: 60px;
    padding: 8px;
    text-align: center;
    text-decoration: none;
    width: 60px;
    margin-bottom: 5px;
}
