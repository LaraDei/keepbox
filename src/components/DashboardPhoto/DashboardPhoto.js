import React, {Component} from 'react'
import Context from '../../context'
import {findPhoto} from '../../helpers'
import './DashboardPhoto.css'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTrash } from'@fortawesome/free-solid-svg-icons'
import AlbumApiService from '../../services/album-api-service'
import TokenService from '../../services/token-service'

export default class DashboardPhoto extends Component{
    static defaultProps = {
        match: {
          params: {}
        },
        history: {
            goBack: () => { }
          },
      }

    static contextType = Context

    handlePhotoDelete= e => {
        e.preventDefault()
        const {photoId} = this.props.match.params
        AlbumApiService.deletePhoto({id: photoId})
        .then(res => {
          this.context.deletePhoto(photoId)
          this.props.history.goBack()
        })
        .catch(error => {
          console.error('delete photo ',{ error })
        })
      }

    render(){
        const { photos=[] } = this.context
        const { photoId } = this.props.match.params
        const photo = findPhoto(photos, photoId) || { content: '' }
        return(
            <div className='DashboardPhoto'>
                <button className="backbtn" onClick={() => this.props.history.goBack()}>Back</button>
                <div className='photo-large'>
                    <img  src={photo.file_location} alt={photo.caption}/>
                </div>
                <div className='photo-details'>
                    <p>{photo.caption}</p>
                    <p>{photo.summary}</p>
                    <p>{photo.date_created}</p>
                </div>
                {TokenService.hasAuthToken()
                ?<button className="deletebtn" onClick={this.handlePhotoDelete}><FontAwesomeIcon icon={faTrash} /> Delete</button>
                : null}
            </div>
        )
    }
}
