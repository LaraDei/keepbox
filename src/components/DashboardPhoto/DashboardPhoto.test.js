
import React from 'react'
import ReactDOM from 'react-dom'
import DashboardPhoto from './DashboardPhoto'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<DashboardPhoto/>, div)
  ReactDOM.unmountComponentAtNode(div)
})
