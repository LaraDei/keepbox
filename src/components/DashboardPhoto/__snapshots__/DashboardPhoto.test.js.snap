// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a .Album by default 1`] = `
<div
  className="DashboardPhoto"
>
  <div
    className="photo-large"
  >
    <img />
  </div>
  <div
    className="photo-details"
  >
    <p />
    <p />
    <p />
  </div>
</div>
`;
